### ISSUE TITLE
Import/maintain list of US stock tickers + prices

### ISSUE THREAD

**Issue by Shpigford** [maybe-finance/maybe#166]
## High level summary
We need to import all US stock tickers and then keep prices updated on an ongoing basis. This is foundational for future feature work.

There's code already written that uses Polygon.io for US stock data and that should be fine to get us started as they have a free plan for individual use.

A good bit of this is already done in the codebase, but it was based around only pulling in necessary tickers based on Plaid + Finicity imports.

We're about to allow people to manually enter any stock and have it track it, so we need to rework things a bit.

## Functional notes/scope
* Relevant database tables/models: Security, SecurityPricing
* A listing of all symbols: https://polygon.io/docs/stocks/get_v3_reference_tickers
* Stock data import will need to happen either on app setup or some sort of check each time the app is booted
* A daily job needs to be run to check for new tickers
* Should pull as much historical pricing data as Polygon allows for a given API key (they have different pricing plans if folks want to pay for more data). This is then stored in SecurityPricing.
* We should think about allowing for "real time" pricing. The Polygon free plan only has "daily" pricing data, but if you're on a paid plan you can get more updated data. Need some sort of background job for this.
* We need to track what stock exchange a given symbol is associated with (I don't believe this is currently tracked in Security)
* As far as database modeling goes, this should err towards flexibility to add additional stock exchanges and tickers in the near future.
* There's no frontend component to this. It's purely about establishing available data for use in other features.
* This will be used by other parts of the app, such as #131 

## Bounty
There's a $250 bounty for this.

**Comment by mrkirthi-24** [comment]
Hi @Shpigford. Not able to login.

![Screenshot 2024-01-20 at 3 12 06 PM](https://github.com/maybe-finance/maybe/assets/53830546/781c97de-dbeb-4cb8-bf9a-3c8a61162602)

![Screenshot 2024-01-20 at 3 09 13 PM](https://github.com/maybe-finance/maybe/assets/53830546/306134f9-9acd-45e1-b824-81dd7e5182b3)

**Comment by mrkirthi-24** [comment]
```
event - compiled client and server successfully in 917 ms (2693 modules)
input parse error ZodError: [
  {
    "code": "invalid_union_discriminator",
    "options": [
      "main",
      "sidebar"
    ],
    "path": [
      "flow"
    ],
    "message": "Invalid discriminator value. Expected 'main' | 'sidebar'"
  }
]
    at handleResult (/Users/hercules/Desktop/maybe/node_modules/zod/lib/types.js:29:23)
    at ZodDiscriminatedUnion.safeParse (/Users/hercules/Desktop/maybe/node_modules/zod/lib/types.js:140:16)
    at ZodDiscriminatedUnion.parse (/Users/hercules/Desktop/maybe/node_modules/zod/lib/types.js:120:29)
    at EndpointFactory.<anonymous> (/Users/hercules/Desktop/maybe/dist/apps/server/webpack:/maybe-finance/libs/server/shared/src/endpoint.ts:47:36)
    at Generator.next (<anonymous>)
    at /Users/hercules/Desktop/maybe/node_modules/tslib/tslib.js:117:75
    at new Promise (<anonymous>)
    at Object.__awaiter (/Users/hercules/Desktop/maybe/node_modules/tslib/tslib.js:113:16)
    at /Users/hercules/Desktop/maybe/dist/apps/server/webpack:/maybe-finance/libs/server/shared/src/endpoint.ts:44:41
    at newFn (/Users/hercules/Desktop/maybe/node_modules/express-async-errors/index.js:16:20) {
  issues: [
    {
      code: 'invalid_union_discriminator',
      options: [Array],
      path: [Array],
      message: "Invalid discriminator value. Expected 'main' | 'sidebar'"
    }
  ],
  addIssue: [Function (anonymous)],
  addIssues: [Function (anonymous)],
  errors: [
    {
      code: 'invalid_union_discriminator',
      options: [Array],
      path: [Array],
      message: "Invalid discriminator value. Expected 'main' | 'sidebar'"
    }
  ]
}
```

**Comment by mrkirthi-24** [comment]
Given my lack of expertise with TSDB, I'd like to delegate this issue to @karan0handa. I appreciate the opportunity, @Shpigford. I'm eager to contribute to the frontend and will proudly do so until I gain proficiency in the backend in the upcoming days. Thank you!

**Comment by tmyracle** [comment]
Closed by #213
